### DISCUSSION TITLE
ESO - Tracking abilities with multiple effects

### DISCUSSION THREAD

**Discussion by MizzerMage** [DakJaniels/FancyActionBarPlus#30] [Q&A] [answered]
Hello - Would you please explain how to configure Fancy Action bars to prioritize the shorter of two ability effects and then continue with the remainder of the longer after the 1st expires?  Specifically, i would like to track the Resolving Vigor 5 sec heal (ID 61509) and then the remaining 15 seconds of the 20 sec Minor Resolve (ID 61693).

**Comment by ACastanza** [comment, marked as answer]
Hello,

By default FAB+ will only track one effect for a given skill. This is intended to create certainty about what your action bar is showing,
That said it might be possible to do this depending on what the basegame behavior of this still is.

By default Resolving Vigor is configured to track Minor Resolve

`  [61507] = { 61693 }; -- resolving vigor` in the config.lua

Configuring it to track effect ID 61509 would get it to show the heal, but this would disable minor resolve tracking.

However, due to a quirk in the basegame ability bar timers for Vigor what you can do is this;

Open up the config.lua file and delete resolving vigor's config from L236:

![image](https://github.com/user-attachments/assets/b9329854-86b3-45f7-a052-8e24ce992c1d)

Once you've done that, in game configure vigor to track the 61509 Heal ID in ABILITY CONFIGURATION > TRACKED EFFECTS like so:

![image](https://github.com/user-attachments/assets/f76af033-1654-4359-9a3d-57d9762a1376)


Now, under "ADDITIONAL TRACKING OPTIONS" > Miscellaneous Options, at the bottom, turn "Allow Fallback Timers" to "ON"

![image](https://github.com/user-attachments/assets/a8bab999-b692-47d2-a729-012f1305baac)

Now the Resolving Vigor slot will track the heal, and then switch to tracking ZOS's default timer, which happens to be the duration of the Minor Resolve.

Note that when you update Fancy Action Bar+ you would need to delete the line from the Config.lua again, although the rest of the changes should stick and not need any additional adjustments besides that.

I might make this the default behavior in a future release since I can see people wanting to do this.


Let me know if it's still not working for you after following these steps

**Reply by MizzerMage** [reply]
Thank you so much for taking the time to respond with such detailed instructions!  Working as described.  Awesome!  

Note:  I needed add a step to the process to accomplish, I think since the ID has already been changed to track the heal:

Since I had already configured Resolving Vigor to track the 5 second HoT before you responded, I followed your instruction to first comment out the line 236 but skipped the Add New ID step since I had already completed (My change appeared in the Saved Changes dropdown list) so I jumped directly to toggling ON the the Allow Feedback Timer.  After a ReloadUI, I tested but this did not cause the remaining duration of the default minor resolve timer to continue to countdown.  So, I Reset my saved Resolving Vigor change in the Info > Tracked Effects panel and then followed your process and it worked!  Not sure why I had to delete my initial saved change but thought to create the exact same conditions as your scenario and tried again.

**Reply by ACastanza** [reply]
Awesome! I'll look into why you needed to first reset the skill config. That shouldn't have been the case. Possibly an inconsistent state was triggered. It might've also been fixed by a reloadui. Regardless, I'll look into it. For now though I've closed this as resolved. Feel free to open a new discussion thread if you encounter any other issues/find anything else that seems to not be working as intended (or as you would like).

**Reply by MizzerMage** [reply]
I really appreciate the what you and others have done with AddOns for ESO.  I have reviewed code and started to learn to create an AddOn as well but got stuck with managing and tracking events, understanding required variables to pass to APIs to return desired results, and general debugging techniques.

Any advise for a novice dev that would like to learn?

Thanks again for your help!

**Reply by ACastanza** [reply]
My two biggest pieces of advice are to shamelessly "take inspiration" from other addons, and to abuse the hell out of d() debugging statements to print things to chat until things start working. Find an addon that seems like it might be even vaguely related to the thing you want your addon to do or even just interacts with the same game events, look through it's code and see if you can figure out how it works, enable any debugging they baked in, add your own d() statements if it doesn't. And also, if you're new to LUA, ChatGPT is actually a surprisingly good debugging tool, it's code isn't great, but it can definitely help if you're stuck.

**Reply by ACastanza** [reply]
Okay so this appears to not be sticking after a reloadui/relog not sure why. very odd.
Edit: I have a fix for it and it'll be in the next release

**Reply by MizzerMage** [reply]
Cool.  I noticed it as well.  At first I thought because I got a buff in IA that extended Major/Minor buffs by a % but then noticed it continued to drop after I reconfigured and was dropping on multiple toons.  Glad to hear you have a fix.  Thanks!
